parquet-column/src/main/java/org/apache/parquet/column/page/DataPageV2.java

```diff
@@ -163,6 +170,33 @@ public DataPageV2(
     this.isCompressed = isCompressed;
   }
 
+  public DataPageV2(
+      int rowCount,
+      int nullCount,
+      int valueCount,
+      BytesInput repetitionLevels,
+      BytesInput definitionLevels,
+      Encoding dataEncoding,
+      BytesInput data,
+      int compressedSize,
+      int uncompressedSize,
+      Statistics<?> statistics,
+      boolean isCompressed) {
+    super(compressedSize, uncompressedSize, valueCount);
+    if (!isCompressed && compressedSize != 0) {
+      throw new IllegalArgumentException("compressedSize must be 0 if page is not compressed");
+    }
```

Comment:
In Parquet C++, compressed_size is always written out.

Reply:
So we only should set the `is_compressed` as false and don't care about the `compressed_size` for this case?

Reply:
IMO, we should assume `compressedSize` == `uncompressedSize` when it is uncompressed. `UNCOMPRESSED` codec is still a valid compression codec type, otherwise, `RowGroup.total_compressed_size` may have problems.

Reply:
> IMO, we should assume `compressedSize` == `uncompressedSize` when it is uncompressed. 

No, because `compressedSize` can also include the overhead of encryption.

`uncompressedSize` is sizeof(levels + uncompressed data).
`compressedSize` is sizeof(encrypted(levels + compressed data))

Reply:
Ah yes, I forgot encryption.

Reply:
I reverted here.